src/renderer/qt/qt_window_manager.cc

```diff
@@ -339,13 +339,67 @@ void FillCandidates(const commands::Candidates &candidates,
   const int width = kColumn0Width + max_width1 + max_width2 + kColumn3Width;
   table->resize(width, total_height);
 }
+
+class VirtualRect {
+ public:
+  static VirtualRect FromNativeRect(const Rect &native_rect) {
+    for (const QScreen *screen : QGuiApplication::screens()) {
+      const Rect rect = TranslateToVirtual(screen, native_rect);
+      const QRect screen_rect = screen->geometry();
+
+      // Use top left to locate a screen
+      if (screen_rect.contains(rect.Left(), rect.Top())) {
+        return VirtualRect(rect, mozc::renderer::GetRect(screen->geometry()));
+      }
+    }
+
+    // fall back to primary screen
+    // TODO: Return the nearest monitor rect instead.
+    // See GetMonitorRect
+    const QScreen *screen = QGuiApplication::primaryScreen();
+    const Rect rect = TranslateToVirtual(screen, native_rect);
+    return VirtualRect(rect, mozc::renderer::GetRect(screen->geometry()));
+  }
+
+  Rect GetRect() const { return rect_; }
+
+  Rect GetMonitorRect() const { return monitor_rect_; }
+
+ private:
+  VirtualRect(const Rect &rect, const Rect &monitor_rect)
+      : rect_(rect), monitor_rect_(monitor_rect) {}
+
+  static Rect TranslateToVirtual(const QScreen *screen,
+                                 const Rect &native_rect) {
+    const double device_pixel_ratio = screen->devicePixelRatio();
+    // screen_left, screen_top have the same value in both virtual and native
+    // coordinate
+    const int screen_left = screen->geometry().x();
```

Comment:
I guess negative coordinate is not used in Linux in general, but for completeness I'd suggest using `std::floor` so that this method does the uniform transformation even when a negative coordinate appears.

```
#include <cmath>
```

```
    const int screen_left = screen->geometry().x();
    const int screen_top = screen->geometry().y();
    const int dx = native_rect.Left() - screen_left;
    const int dy = native_rect.Top() - screen_top;
    const int vx = static_cast<int>(std::floor(dx)) + screen_left;
    const int vy = static_cast<int>(std::floor(dy)) + screen_top;
```

If you don't use `std::floor` there, both `0.5` and `-0.5` will be rounded into `0.0`, which does not make sense.

Reply:
Thank you. It now uses `floor` to convert.